File: fixtures/scale/dart/large/lib/ciphers/porta/porta_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'porta_cipher.dart';
import 'porta_key.dart';
import 'porta_vectors.dart';

class PortaSuite implements CipherSuite {
  @override
  String name() => 'porta';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = PortaCipher(PortaKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in portaSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/porta/PortaSuite.java
package cryptobench.ciphers.porta;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class PortaSuite implements CipherSuite {
    @Override
    public String name() {
        return "porta";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new PortaCipher(PortaKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : PortaVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}
